connect.go

```diff
@@ -33,12 +34,17 @@ type Connection struct {
 }
 
 type FileMetaData struct {
-	Name string
-	Size int
-	Hash string
-	Path string
+	Name  string
+	Size  int
+	Hash  string
+	Path  string
+	isDir bool
```

Comment:
You should export `isDir` (make it `IsDir`) so it will be Marshalled into the JSON and automatically sent with the Meta data to the receiver. Otherwise looks good :)

Reply:
Thanks for your review =)